src/libraries/Microsoft.Extensions.Logging/gen/TemplateHelper.cs

```diff
@@ -0,0 +1,346 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Collections.Generic;
+using System.Runtime.CompilerServices;
+using System.Text;
+
+namespace System.Text
+{
+    internal static class TemplateHelper
+    {
+        /// <summary>
+        /// Parses the message template and throws exception for malformed messages.
+        /// </summary>
+        internal static string Parse(this string msg, List<string>? templates)
```

Comment:
>Parse [](http://example.com/codeflow?start=31&length=5)

do we really need to throw? can't this converted to TryParse? and let it communicate the parsing results and can throw in the calling site if needed? 

I am just wondering why we need to throw then catch it again?

Reply:
Sure. I'll switch it from throwing exception